### ISSUE TITLE
UnicodeDecodeError with file_diff_outputs

### ISSUE THREAD

**Issue by robintw** [recipy/recipy#121]
When running with the `file_diff_outputs` setting on, I get the following error when running the example script:

```
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/Users/robin/code/recipy/recipy/log.py", line 282, in output_file_diffs
    diff = difflib.unified_diff(open(item['tempfilename']).readlines(),
  File "/Users/robin/anaconda3/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
```

This is on Python 3, using the following script:

```
import recipy
import pandas as pd
from matplotlib.pyplot import *

data = pd.read_csv('data.csv')

data.plot(x='year', y='temperature')
savefig('graph.png')

data.temperature = data.temperature * 10
data.to_csv('output.csv')
```

**Comment by robintw** [comment]
I think you implemented the output file diffing @jvdzwaan  - are you able to have a look at this?

**Comment by jvdzwaan** [comment]
Ah, this probably has something to do with the difference between python 2 and 3. I'll fix it

**Comment by jvdzwaan** [comment]
Okay, this problem is a little trickier than I thought. 

One of the problems is that Python2 uses a different default encoding than python3. Another problem is that doing file diffs for non-text formats does not really make sense.

The solution I currently have in mind is:
1. Guess the mimetype of the output file and only store file diff if it is a text file
2. Use utf-8 as default encoding. If utf-8 doesn't work, issue a warning (so the user can set encodings to utf-8 and rerun the script)

Alternatively, we could try both utf-8 and latin-1 in step 2, I think this will take care of the difference in default encodings in python 2 and 3 (however, I haven't tried it on windows or mac).

What do you think, @robintw ?

**Comment by robintw** [comment]
Argh, string encoding stuff is always so much of a pain!

I think your solution sounds good - my only question is how you would guess the mimetype? Still, if you've got a way of doing that then the overall approach is sound :)

I can test on Window and OS X and see how it works - if you want to work it up into a PR or a separate branch then I'm happy to test.

**Comment by jvdzwaan** [comment]
Using [mimetype](https://docs.python.org/2/library/mimetypes.html). I think it basically looks at the file extension, but probably (hopefully) it is a little more intelligent than that.

**Comment by robintw** [comment]
Fixed by #166.
